src/NUnitFramework/framework/Internal/Commands/OneTimeTearDownCommand.cs

```diff
@@ -69,7 +69,7 @@ public override TestResult Execute(TestExecutionContext context)
                         item.RunTearDown(context);
 
                 IDisposable disposable = context.TestObject as IDisposable;
-                if (disposable != null)
+                if (disposable != null && Test is TestFixture)
```

Comment:
What if it's a SetUpFixture? Ideally, we should have a common base class for all those tests that are represented by Types we must create. For now, I'd add a check for SetUpFixture and maybe a TODO comment.